scheduler/pkg/kafka/config/tls.go

```diff
@@ -79,27 +82,46 @@ func setupSASLSSLAuthentication(config kafka.ConfigMap) error {
 	// see https://github.com/confluentinc/confluent-kafka-go/issues/827 (Fixed needs updating and testing in our code)
 
 	config["ssl.ca.location"] = caCert.CaPath
+
+	endpointMechanism := tls.GetEndpointIdentificationMechanismFromEnv(tls.EnvSecurityPrefixKafkaClient)
+	config["ssl.endpoint.identification.algorithm"] = endpointMechanism
 	return nil
 }
 
 func setupTLSAuthentication(config kafka.ConfigMap) error {
-	cs, err := tls.NewCertificateStore(tls.Prefix(EnvKafkaClientPrefix),
-		tls.ValidationPrefix(EnvKafkaBrokerPrefix))
+	var cs *tls.CertificateStore
+	cs, err := tls.NewCertificateStore(
+		tls.Prefix(EnvKafkaClientPrefix),
+		tls.ValidationPrefix(EnvKafkaBrokerPrefix),
+	)
 	if err != nil {
-		return err
+		cs, err = tls.NewCertificateStore(
+			tls.ValidationOnly(true),
+			tls.ValidationPrefix(EnvKafkaBrokerPrefix),
+		)
```

Comment:
Just to clarify, does this bit try to first set up mTLS and then falls back to one-sided TLS after erroring?

Reply:
yes - exactly

Reply:
Got it - it may be worth adding a comment to clarify that's what's going on?